### PR TITLE
don't set session if redirected not set

### PR DESCRIPTION
**Pull request by karlhungus**
@Thibaut @qq99 

##### Problem 

I found an issue in our usage of this where redirecting on location.replace would set `session[:_turbolinks_redirect_to]` without headers. This would cause subsequent requests to get redirected when they shouldn't

https://github.com/Shopify/shopify/issues/35973#issuecomment-69338390

##### Solution

Update to latest version of `xhr_headers`: https://github.com/rails/turbolinks/blob/b7553314d82f9907ea6da57361bb4fe9e956eaf6/lib/turbolinks/xhr_headers.rb#L28. I followed the history of xhr_headers, and all changes seem reasonable. 

I have yet to double check this fixes my actual problem :).

### CODE REVIEW

**Comment by Thibaut** [comment]
cc @arthurnn 
will look at this around noon

**Comment by Thibaut** [comment]
This needs a good :tophat: but otherwise :+1:.

**Comment by qq99** [comment]
want to make a PR ref'ing this SHA ?

**Comment by karlhungus** [comment]
> want to make a PR ref'ing this SHA ?

Will do, 

@qq99 any notion why `PartialPageRefreshTest#test_when_I_use_an_XHR_and_POST_to_an_endpoint_that_returns_me_a_302,_I_should_see_the_URL_reflecting_that_redirect_too` wouldn't be setting `X-XHR-Referer` -- I put breakpoints in ['tubolinks.coffee'](https://github.com/Shopify/turbograft/blob/098ead7d84ff22b9f4350346e292f99e3ab55f33/lib/assets/javascripts/turbograft/turbolinks.coffee#L81) but they weren't triggered

**Comment by qq99** [comment]
Is `fetchReplacement` being called?

**Comment by kurtfunai** [comment]
Just going to mention that this seems related to issue #16 / PR #15

**Comment by karlhungus** [comment]
@qq99 noted:
https://github.com/Shopify/turbograft/blob/098ead7d84ff22b9f4350346e292f99e3ab55f33/lib/assets/javascripts/turbograft/remote.coffee#L22
we need to set header in here too I think and then `tg-remote` should be OK

**Comment by karlhungus** [comment]
Fixes: https://github.com/Shopify/shopify/issues/35973

**Comment by arthurnn** [comment]
We should test this repo against multiple rails versions! we patch rails it self, and try to be compatible with multiple versions, so we need to tests those!

**Comment by karlhungus** [comment]
>We should test this repo against multiple rails versions!

I'll add that too

**Comment by karlhungus** [comment]
re-ping @arthurnn @Thibaut @qq99 /cc @Shopify/admin 

This adds the multiple rails versions (kind of guessed here), fixes the test by adding the header forceably. Updates tg-remote to make use of the referer (not sure if that was ok).

If this is all-right, i'll merge https://github.com/Shopify/shopify/pull/36047 if things go awry then i'll revert.

**Comment by qq99** [comment]
looks legit :+1:

**Comment by arthurnn** [comment]
:+1:

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,12 @@
 language: ruby
 rvm:
-  - "2.0.0"
+  - '2.0.0'
+  - '2.1.2'
```
Comment (Thibaut) [inline .travis.yml]: `2.2.0` ?

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 before_script:
   - sh -e /etc/init.d/xvfb start
   - bundle install
 env:
-  - DISPLAY=":99.0"
+  - DISPLAY=':99.0' RAILS_VERSION='3.1.0'
+  - DISPLAY=':99.0' RAILS_VERSION='3.2.0'
+  - DISPLAY=':99.0'  RAILS_VERSION='4.1.0'
```
Comment (Thibaut) [inline .travis.yml]: `4.2.0`?

**Comment by karlhungus** [comment]
Any rails version/ruby versions i should remove (with all of these builds take a long time and are a bit more flakey)?

**Comment by kurtfunai** [comment]
@karlhungus I might be wrong, but I think that rails 3.1 is no longer supported?

Found an old article, http://weblog.rubyonrails.org/2013/2/24/maintenance-policy-for-ruby-on-rails/
Security issues:
>Severe security issues:
>For severe security issues we will provide new versions as above, and also the last major release series will receive patches and new versions. The classification of the security issue is judged by the core team.
>Currently included series: 3.2.z, 3.1.z, 2.3.z
>After the Rails 4 release: 4.0.z, 3.2.z

**Comment by karlhungus** [comment]
Thanks @kurtfunai , 3.1 removed :). Cool to merge this -- then i'll create 0.1.6?

**Comment by qq99** [comment]
merge :+1:

**Pull request merged**